validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -382,6 +385,103 @@ def id_by_index(index, resources):
             return resources[index].address
 
 
+class _Sorter(object):
+    """A static class containing a method to sort lists of resources based on
+    SortControls sent with the request.
+    """
+    @classmethod
+    def sort_resources(cls, request, resources, fail_enum, header_proto=None):
+        """Sorts a list of resources based on a list of sort controls
+
+        Args:
+            request (object): The parsed protobuf request object
+            resources (list of objects): The resources to be sorted
+            fail_enum (int, enum): The enum status to raise with invalid keys
+            header_proto(class): Class to decode a resources header
+
+        Returns:
+            list: The sorted list of resources
+        """
+        if not request.sorting:
+            return resources
+
+        value_handlers = [cls._ValueHandler(controls, fail_enum, header_proto)
+                          for controls in request.sorting]
+
+        def sorter(resource_a, resource_b, value_handlers=value_handlers):
+            if not value_handlers:
```

Comment:
This threw me off because I didn't know an empty list was false-y in python. Is this good practice? Is it better than `if len(value_handlers) < 1:`?